zigzag/classes/hardware/architecture/ImcArray.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from typing import Dict
+if __name__ == "__main__":
+    from dimension import Dimension
+    from DimcArrayUnit import DimcArrayUnit
+    from AimcArrayUnit import AimcArrayUnit
+else:
+    from zigzag.classes.hardware.architecture.dimension import Dimension
+    from zigzag.classes.hardware.architecture.DimcArrayUnit import DimcArrayUnit
+    from zigzag.classes.hardware.architecture.AimcArrayUnit import AimcArrayUnit
+
+
+class ImcArray:
```

Comment:
Same comment as for the CostModelEvaluation: If you use inheritance you can build on top of the original "OperationalArray" object and don't have to redefine the dimensions attributes.